packages/@expo/prebuild-config/src/plugins/icons/withIosIcons.ts

```diff
@@ -21,31 +21,92 @@ export const withIosIcons: ConfigPlugin = (config) => {
   ]);
 };
 
-export function getIcons(config: Pick<ExpoConfig, 'icon' | 'ios'>): string | null {
-  // No support for empty strings.
-  return config.ios?.icon || config.icon || null;
+export function getIcons(config: Pick<ExpoConfig, 'icon' | 'ios'>): IOSIcons | string | null {
+  const iosSpecificIcons = config.ios?.icon;
+
+  if (iosSpecificIcons) {
+    // For backwards compatibility, the icon can be a string
+    if (typeof iosSpecificIcons === 'string') {
+      return iosSpecificIcons || config.icon || null;
+    }
+
+    // in iOS 18 introduced the ability to specify dark and tinted icons, which users can specify as an object
+    if (!iosSpecificIcons.any && !iosSpecificIcons.dark && !iosSpecificIcons.tinted) {
+      return config.icon || null;
+    }
+
+    return iosSpecificIcons;
+  }
+
+  if (config.icon) {
+    return config.icon;
+  }
+
+  return null;
 }
 
 export async function setIconsAsync(config: ExpoConfig, projectRoot: string) {
   const icon = getIcons(config);
-  if (!icon) {
+
+  if (icon === null) {
     WarningAggregator.addWarningIOS('icon', 'No icon is defined in the Expo config.');
+  } else if (typeof icon === 'string') {
+    if (!icon) {
+      WarningAggregator.addWarningIOS('icon', 'No icon is defined in the Expo config.');
+    }
+  } else if (typeof icon === 'object') {
+    if (!icon?.any && !icon?.dark && !icon?.tinted) {
+      WarningAggregator.addWarningIOS('icon', 'No icon is defined in the Expo config.');
+    }
   }
```

Comment:
These conditions could be combined, no point having so many separate conditions with the same line of executing code.
```suggestion
  if (!icon || (typeof icon === 'string' && !icon) || (typeof icon === 'object' && !icon?.any && !icon?.dark && !icon?.tinted)) {
    WarningAggregator.addWarningIOS('icon', 'No icon is defined in the Expo config.');
  }
```
Hope work gets done in this area soon or this PR gets merged. Thanks for your work :)